### DISCUSSION TITLE
-override not reflecting for integer parameter

### DISCUSSION THREAD

**Discussion by ap766** [OpenModelica/OpenModelica#11467] [Q&A]
1)So here I have N_Trays which is a integer parameter.PC a real parameter

![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/78c782f4-ec3a-4940-bf83-d02419aaef37)

![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/fce1fec6-9168-4cc2-ba06-edd05eb99637)

2)I am using this command in the GUI ,
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/ce7b45de-e939-40b9-911c-3c6577027af2)

3)As you can see the resulting .mat file. The changes are reflected for PC but not for N_Trays.
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/92fa0684-e04e-4abe-afb2-0058ed6ec946)

Does it have to do with it being an integer? Can you please help me resolve this?

**Comment by adrpo** [comment]
The problem seems to be that `N_Trays` is a calculated parameter which means that even if you override it, it will be calculated again during initialization. Change the parameters that are used to calculate `N_Trays` instead.

**Reply by ap766** [reply]
oh okay! thank you. 
But actually these have been declared the same way as the others parameters in the openmodelica code ? They are not calculated as it shows in the XML file. 

![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/3f43dbed-a882-4442-a329-0639935516c5)

**Reply by Arinomo** [reply]
@ap766 did you define size of an array using `N_Trays`?  if so, than `N_Trays` is a structural parameter and IFAIK it could not be overridden.
